### ISSUE TITLE
`decode_bytes()` improper error handling

### ISSUE THREAD

**Issue by mih** [datalad/datasalad#35]
I have a line from `git ls-files` that comes out of `itemize()` like so:

```
b'120000 39d6af4c8d20bfdd8effecf5babaaacc971873e6 0\tBob Marley & The Wailers/Live 9-23-80/08 War \xaf No More Trouble.shn.mp3'
```

`decode_bytes()` trips on the `\xaf` and yields

```
'120000 39d6af4c8d20bfdd8effecf5babaaacc971873e6 0\tBob Marley & The Wailers/Live 9-23-80/08 War \\xaf'
```

i.e. it swallows the end of the file name.

A test like the following is sufficient to reproduce the problem

```diff
diff --git a/datalad_next/itertools/tests/test_decode_bytes.py b/datalad_next/itertools/tests/test_decode_bytes.py
index a463cc4..ee5517f 100644
--- a/datalad_next/itertools/tests/test_decode_bytes.py
+++ b/datalad_next/itertools/tests/test_decode_bytes.py
@@ -35,3 +35,9 @@ def test_no_empty_strings():
     # check that empty strings are not yielded
     r = tuple(decode_bytes([b'\xc3', b'\xb6']))
     assert r == ('ö',)
+
+
+def test_marley():
+    beits = b'08 War \xaf No More Trouble.shn.mp3'
+    r = tuple(decode_bytes([beits]))
+    assert r[0] == '08 War \xaf No More Trouble.shn.mp3'
```

Giving:

```
>       assert r[0] == '08 War \xaf No More Trouble.shn.mp3'
E       AssertionError: assert '08 War \\xaf' == '08 War ¯ No ...ouble.shn.mp3'
E         
E         - 08 War ¯ No More Trouble.shn.mp3
E         + 08 War \xaf
```

(the test target is probably not the real thing, but it shows the problematic behavior)

**Comment by mih** [comment]
Fixed by #36

**Comment by mih** [comment]
Reopening, because the docs are not yet adjusted
